components/Heroes/Benchmark.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import CircularProgress from 'material-ui/CircularProgress';
+
+import style from './heroes.css';
+import { REDUCER_KEY } from './../../reducers';
+import { getBenchmark } from './../../actions';
+
+import BenchmarkTable from './BenchmarkTable';
+import BenchmarkBadge from './BenchmarkBadge';
+
+class Benchmark extends Component {
+
+  componentDidMount() {
+    if (this.props.routeParams.hero_id) {
+      this.props.getBenchmark(this.props.routeParams.hero_id);
+    }
+  }
+
+  render() {
+    const { isLoading, isDone, heroes, result, heroId } = this.props;
+
+    return (
+      <div>
+        {isLoading ?
+          <div className={style.Loading}>
+            <CircularProgress color="#fff" />
+          </div> : ''}
+
+        {isDone ? <div>
+          <BenchmarkBadge hero={heroes[heroId]} />
+          <BenchmarkTable data={result} />
+        </div> : ''}
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = (state) => ({
+  heroes: state[REDUCER_KEY].heroes,
+  heroId: state[REDUCER_KEY].gotBenchmark.hero_id,
+  result: state[REDUCER_KEY].gotBenchmark.result,
+  isLoading: state[REDUCER_KEY].gotBenchmark.loading,
+  isDone: state[REDUCER_KEY].gotBenchmark.done,
```

Comment:
yes, `isDone` needed by this component, I was tried to use `isLoading`, the problem with `isLoading` once you load the page, first initial state is `false`, however `result` initial state is an empty object, checking whether object is empty or not cost more than using `isDone` and I don't want to check whether an object is empty or not because in my opinion it will make code slightly unpleasant to read.